File: src/ui/gfx/Camera.java
package ui.gfx;

import gameplay.Point;
import gameplay.Rectangle;

import static ui.gfx.Renderer.*;

public class Camera {
    private Point point;

    public Camera() {
        this.point = new Point();
    }

    public Camera(int x, int y) {
        this.point = new Point(x, y);
    }

    public void cap(Rectangle borders, int windowWidth, int windowHeight) {
        this.point = new Point(
                point.capX(-CAMERA_SIDE_MARGIN, borders.width - windowWidth + CAMERA_SIDE_MARGIN),
                point.capY(-CAMERA_BOTTOM_MARGIN, borders.height - windowHeight + CAMERA_TOP_MARGIN)
        );
    }

    public int getX() {
        return this.point.getX();
    }

    public int getY() {
        return this.point.getY();
    }
}

File: src/main/kotlin/ui/gfx/Camera.kt
package ui.gfx

import gameplay.Point
import gameplay.geometry.Rectangle

class Camera(private var point: Point) {

    constructor() : this(Point())
    constructor(x: Int, y: Int) : this(Point(x.toDouble(), y.toDouble()))

    fun cap(borders: Rectangle, windowWidth: Int, windowHeight: Int) {
        point = Point(
                point.capX(-Renderer.CAMERA_SIDE_MARGIN.toDouble(), borders.width - windowWidth + Renderer.CAMERA_SIDE_MARGIN),
                point.capY(-Renderer.CAMERA_BOTTOM_MARGIN.toDouble(), borders.height - windowHeight + Renderer.CAMERA_TOP_MARGIN)
        )
    }

    val x: Int
        get() = point.x.toInt()

    val y: Int
        get() = point.y.toInt()
}
